src/addons/filter/multilang2/services/handlers/multilang2.ts

```diff
@@ -0,0 +1,114 @@
+
+import { Injectable } from '@angular/core';
+
+import { CoreLang } from '@services/lang';
+import { CoreFilterDefaultHandler } from '@features/filter/services/handlers/default-filter';
+import { makeSingleton } from '@singletons';
+
+/**
+ * Handler to support the Multilang filter.
+ */
+@Injectable({ providedIn: 'root' })
+export class AddonFilterMultilang2HandlerService extends CoreFilterDefaultHandler {
+
+    name = 'AddonFilterMultilang2Handler';
+    filterName = 'multilang2';
+    replacementDone = false;
+
+    /**
+     * Cache of parent language(s) of a given language.
+     */
+    protected parentCache: string[] = [];
+
+    /**
+     * This function filters the received text based on the language
+     * tags embedded in the text, and the current user language or
+     * 'other', if present.
+     *
+     * @param text The text to filter.
+     * @returns string The filtered text for this multilang block.
+     */
+    async filter(text: string): Promise<string> {
+        const currentLang = await CoreLang.getCurrentLanguage();
+
+        if (text.indexOf('mlang') === -1) {
+            return text;
+        }
+
+        this.replacementDone = false;
+
+        if (!this.parentCache[currentLang] && CoreLang.getParentLanguage(currentLang)) {
+            const parentlangs = CoreLang.getParentLanguage(currentLang);
+            this.parentCache[currentLang] = parentlangs;
+        }
+
+        const search = /{\s*mlang\s+((?:[a-z0-9_-]+)(?:\s*,\s*[a-z0-9_-]+\s*)*)\s*}(.*?){\s*mlang\s*}/g;
+        const result = text.replace(
+            search,
+            (substring, language, content) => this.replaceLangs(currentLang, [substring, language, content]),
+        );
+
+        if (result === null) {
+            return text; // Error during regex processing, keep original text.
+        }
+
+        if (this.replacementDone) {
+            return result;
+        }
+
+        const result2 = text.replace(
+            search,
+            (substring, language, content) => this.replaceLangs('other', [substring, language, content]),
+        );
+
+        return result2 ?? text;
+    }
+
+    /**
+     * This function filters the current block of multilang tag. If
+     * any of the tag languages (or their parent languages) match the
+     * specified filtering language, it returns the text of the
+     * block. Otherwise it returns an empty string.
+     *
+     * @param replacelang A string that specifies the language used to filter the matches.
+     * @param langblock An array containing the matching captured pieces of the
+     * regular expression. They are the languages of the tag, and the text associated with those languages.
+     *
+     * @returns replaced string.
+     */
+    protected replaceLangs(replacelang: string, langblock: string[]): string {
+        // Normalize languages.
+        const blocklangs = langblock[1].replace(/-|_/g, '').toLowerCase().split(',');
+        const blocktext = langblock[2];
```

Comment:
```suggestion
        const blocklangs = (langblock[1] ?? '').replace(/-|_/g, '').toLowerCase().split(',');
        const blocktext = langblock[2] ?? '';
```

Right now this isn't needed, but IMO this way the code won't crash if the code changes and this parameters are no longer supplied, because TS wouldn't detect that.